forte/data/data_store.py

```diff
@@ -250,6 +251,46 @@ def _new_group(self, type_name: str, member_type: str) -> List:
         entry += len(self._type_attributes[type_name]) * [None]
         return entry
 
+    def _is_subclass(self, type_name: str, cls) -> bool:
+        r"""This function takes a fully qualified ``type_name`` class name,
+        ``cls`` class and returns whether ``type_name``  class is the``cls``
+        subclass or not.
+
+        Args:
+            type_name: A fully qualified name of an entry class
+            cls: An entry class.
+
+        Returns:
+            A boolean value whether ``type_name``  class is the``cls``
+            subclass or not.
+
+        """
+        if type_name not in self._type_attributes:
+            self._type_attributes[type_name] = {}
+        if "parent_class" not in self._type_attributes[type_name]:
+            self._type_attributes[type_name]["parent_class"] = set()
+        cls_qualified_name = get_full_module_name(cls)
+        type_name_parent_class = self._type_attributes[type_name][
+            "parent_class"
+        ]
+        if len(type_name_parent_class) > 0:
+            return bool(cls_qualified_name in type_name_parent_class)
```

Comment:
If `len(type_name_parent_class) > 0`, I believe it does not necessarily mean that the set contains all possible parents, since you only add one class in line 281 and returns.

Reply:
I guess a simple approach is to check:

```
if cls_qualified_name in type_name_parent_class:
  return True
else:
  # check is_subclass
```

Basically, we are caching the result for this pair, and if not, we can use `is_subclass`. @qinzzz  what do you think